examples/tip-assert/src/main/scala/example/PetService.scala

```diff
@@ -0,0 +1,39 @@
+package example
+
+import scalaj.http.Http
+import io.circe.generic.auto._
+import io.circe.parser._
+import scala.concurrent.{ExecutionContext, Future}
+
+case class Pet(name: String)
+case class PetCreateResponse(id: Long, name: String)
+case class PetGetResponse(id: Long, name: String)
+
+object PetService {
+  def createPet(pet: Pet)(implicit executionContext: ExecutionContext): Future[Long] = Future {
+    val response = Http("https://petstore.swagger.io/v2/pet")
+      .header("Content-Type", "application/json")
+      .header("accept", "application/json")
+      .postData(
+        s"""
+           |{
+           |  "name": "${pet.name}"
+           |}
+        """.stripMargin)
+      .asString
+      .body
```

Comment:
if this were actual production code, slightly better practice would be to wrap this in a `blocking { ... }` so that the execution context can fork another thread if necessary.  I'm not sure if scalaj has a non blocking version.